src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -666,21 +676,34 @@ public static void joinGame(final GameDescription description, final Messengers
         return;
       }
     }
-    joinGame(description.getPort(), description.getHostedBy().getAddress().getHostAddress(), newClassPath, messengers);
+    joinGame(description.getPort(), description.getHostedBy().getAddress().getHostAddress(),
+        areSameVersion ? "" : newClassPath, messengers);
   }
 
   // newClassPath can be null
   private static void joinGame(final int port, final String hostAddressIP, final String newClassPath,
       final Messengers messengers) {
     final List<String> commands = new ArrayList<>();
-    ProcessRunnerUtil.populateBasicJavaArgs(commands, newClassPath);
-    commands.add("-D" + TRIPLEA_CLIENT_PROPERTY + "=true");
-    commands.add("-D" + TRIPLEA_PORT_PROPERTY + "=" + port);
-    commands.add("-D" + TRIPLEA_HOST_PROPERTY + "=" + hostAddressIP);
-    commands.add("-D" + TRIPLEA_NAME_PROPERTY + "=" + messengers.getMessenger().getLocalNode().getName());
-    final String javaClass = "games.strategy.engine.framework.GameRunner";
-    commands.add(javaClass);
-    ProcessRunnerUtil.exec(commands);
+    final boolean startInNewProcess = !("".equals(newClassPath) && MainFrame.getInstance() != null);
```

Comment:
` !("".equals(newClassPath) && MainFrame.getInstance() != null);`
Can be written more simply as:
`newClassPath.isEmpty() || MainFrame.getInstance() == null`

You'll need to double check that `newClassPath` is not null, or add a null check to that above conditional.